test/__init__.py

```diff
@@ -0,0 +1,250 @@
+
+        # Create a tar file
+        tarball_path = os.path.join(cls.temp_dir.name, 'archive.tar.gz')
+        with tarfile.TarFile.open(tarball_path, 'w:gz') as f:
+            f.add(archive_path, 'archive_dir')
+
+        # Create a zip file
+        zip_path = os.path.join(cls.temp_dir.name, 'archive.zip')
+        with zipfile.ZipFile(zip_path, mode='w') as f:
+            f.write(
+                os.path.join(archive_path, 'file_name.txt'),
+                os.path.join('archive_dir', 'file_name.txt'),
+            )
+
+        # Populate the staged.repos file
+        repos = {
+            'immutable/hash': {
+                'type': 'git',
+                'url': to_file_url(gitrepo_path),
+                'version': '5b3504594f7354121cf024dc734bf79e270cffd3',
+            },
+            'immutable/hash_tar': {
+                'type': 'tar',
+                'url': to_file_url(tarball_path),
+                'version': 'archive_dir',
+            },
+            'immutable/hash_zip': {
+                'type': 'zip',
+                'url': to_file_url(zip_path),
+                'version': 'archive_dir',
+            },
+            'immutable/tag': {
+                'type': 'git',
+                'url': to_file_url(gitrepo_path),
+                'version': 'tags/0.1.27',
+            },
+            'vcs2l': {
+                'type': 'git',
+                'url': to_file_url(gitrepo_path),
+                'version': 'heads/main',
+            },
+            'without_version': {
+                'type': 'git',
+                'url': to_file_url(gitrepo_path),
+            },
+        }
+
+        cls.repos_file_path = os.path.join(cls.temp_dir.name, 'staged.repos')
+        with open(cls.repos_file_path, 'w') as f:
+            yaml.safe_dump({'repositories': repos}, f)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.repos_file_path = None
+        cls.temp_dir and cls.temp_dir.cleanup()
+        cls.temp_dir = None
+
+
+class StagedReposFile2(unittest.TestCase):
+    _svn = which('svn')
+    _svnadmin = which('svnadmin')
+    _hg = which('hg')
+    _hg_env = {
+        **os.environ,
+        'HGUSER': 'vcs2l',
+        'EMAIL': 'vcs2l@example.com',
+    }
+    _commit_date = '2005-01-01T00:00:00-06:00'
+
+    temp_dir = None
+    repos_file_path = None
+
+    @classmethod
+    def setUpClass(cls):
+        if not cls._svn:
+            raise unittest.SkipTest('`svn` was not found')
+        if not cls._svnadmin:
+            raise unittest.SkipTest('`svnadmin` was not found')
+        if not cls._hg:
+            raise unittest.SkipTest('`hg` was not found')
+        try:
+            # check if the svn executable is usable (on macOS)
+            # and not only exists to state that the program is not installed
+            subprocess.check_call([cls._svn, '--version'], stdout=subprocess.DEVNULL)
+        except subprocess.CalledProcessError:
+            raise unittest.SkipTest('`svn` was not found')
```

Comment:
I think this approach would lose the exception context upon raise, based on [W0707](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/raise-missing-from.html), we can include as follows:


```suggestion
        except subprocess.CalledProcessError as exc:
            raise unittest.SkipTest('`svn` was not found') from exc
```

Reply:
Does that context matter in this case? I would be fine NOQA'ing if we're explicitly dropping in favor of the skiptest expression.